wrangler-core/src/main/java/io/cdap/wrangler/utils/Json2Schema.java

```diff
@@ -73,62 +73,75 @@ public Schema toSchema(String id, Row row) throws RecordConvertorException {
       String name = column.getFirst();
       Object value = column.getSecond();
 
-      // First, we check if object is of simple type.
-      if (value instanceof String || value instanceof Integer || value instanceof Long || value instanceof Short ||
-          value instanceof Double || value instanceof Float || value instanceof Boolean || value instanceof byte[]) {
-        try {
-          Schema schema = Schema.nullableOf(new SimpleSchemaGenerator().generate(value.getClass()));
+      if (value instanceof Schema) {
```

Comment:
when would the value be a Schema?

Reply:
yea even I am not sure. But that is part of restructuring of existing code.
https://github.com/data-integrations/wrangler/blob/release/4.0/wrangler-core/src/main/java/io/cdap/wrangler/utils/Json2Schema.java#L127